injector/src/main/java/edu/ucr/cs/riple/injector/Helper.java

```diff
@@ -66,43 +74,57 @@ public static String extractCallableName(String signature) {
           --level;
           break;
         default:
-          if (level == 0) ans.append(current);
+          if (level == 0) {
+            ans.append(current);
+          }
       }
     }
     return ans.toString();
   }
 
+  /**
+   * Checks if callableDec's signature is equals to signature passed in the argument.
+   *
+   * @param callableDec callable declaration node.
+   * @param signature signature of the callable in string.
+   * @return true, if signature matches the callable and false otherwise.
+   */
   public static boolean matchesCallableSignature(
```

Comment:
Definitely not for this PR, but I notice there are a lot of methods here comparing signatures in String format to `CallableDeclaration` by parsing the string multiple times and extracting different portions of it: wouldn't it be more useful to have a single function parsing a string into it's corresponding `CallableDeclaration`? In fact, if the only comparison we need is strict equality, a well crafted `CallableDeclaration.toString()` method could be all we need here, no parsing required.

Reply:
Hi @lazaroclapp, sure that sounds good. However, at the moment I was writing `injector` it wasn't merged into annotator as a sub-module and I was intending to make a robust tool that works with any given signature, not necessarily matching how `error prone` serializes methods signature. It now works with any given input as long as the types and names and correct (simple name or fully qualified name). However, what you mention about parsing it multiple times is a really good point and it worths to address it in this PR. Please find it here where we only parse each signature once. 13ee6712184d74d1edea5e155f3967f5b2c86b57.